pulp_container/app/tasks/sync_stages.py

```diff
@@ -112,33 +152,34 @@ async def run(self):
             relative_url = "/v2/{name}/manifests/{tag}".format(
                 name=self.remote.namespaced_upstream_name, tag=tag_name
             )
-            url = urljoin(self.remote.url, relative_url)
-            downloader = self.remote.get_downloader(url=url)
-            to_download.append(downloader.run(extra_data={"headers": V2_ACCEPT_HEADERS}))
+            tag_url = urljoin(self.remote.url, relative_url)
+            downloader = self.remote.get_downloader(url=tag_url)
+            to_download.append(
+                downloader.run(extra_data={"headers": V2_ACCEPT_HEADERS, "http_method": "head"})
+            )
 
         async with ProgressReport(
             message="Processing Tags",
             code="sync.processing.tag",
             total=len(tag_list),
         ) as pb_parsed_tags:
 
-            for download_tag in asyncio.as_completed(to_download):
-                dl_res = await download_tag
-                with open(dl_res.path, "rb") as content_file:
-                    raw_data = content_file.read()
-                dl_res.artifact_attributes["file"] = dl_res.path
-                saved_artifact = await sync_to_async(_save_artifact_blocking)(
-                    dl_res.artifact_attributes
-                )
+            to_download_artifact = [
+                self._check_for_existing_manifest(download_tag)
+                for download_tag in asyncio.as_completed(to_download)
+            ]
 
-                tag_name = dl_res.url.split("/")[-1]
-                tag_dc = DeclarativeContent(Tag(name=tag_name))
+            for artifact in asyncio.as_completed(to_download_artifact):
+                saved_artifact, content_data, raw_data, response = await artifact
+
+                digest = response.headers.get("docker-content-digest")
```

Comment:
this header won't be necessarily available in all the circumstances.

Reply:
Maybe we could extract this from the `content_data`

Reply:
Hey, I checked this again. It will be available because the `_check_for_existing_manifest` will await for this response and we pass it through if we find an already existing manifest, or generate a new response if we don't find a manifest.

Reply:
i meant that `docker-content-digest` won't necessarily be availalbe, it's not a mandatory header.